keras/backend/tensorflow_backend.py

```diff
@@ -180,30 +180,28 @@ def set_session(session):
 
 # VARIABLE MANIPULATION
 
-def _convert_string_dtype(dtype):
+def _convert_string_dtype(dtype, check_supported_type=True):
```

Comment:
This is a private, internal method, and we could remove it. Consider replacing it with `as_dtype` entirely. Much simpler.

Reply:
Sure! However, some methods only support limited dtypes, so they need to be modified to check type by themselves. Is it OK? Thanks.

Reply:
If done cleanly, then yes.